test/case/infix_interfaces/bridge_vlan_separation.py

```diff
@@ -0,0 +1,171 @@
+            "interfaces": {
+                "interface": [
+                    {
+                        "name": "br0",
+                        "type": "iana-if-type:bridge",
+                        "enabled": True,
+                        "bridge": {
+                            "vlans": {
+                                "vlan": [
+                                    {
+                                        "vid": 10,
+                                        "untagged": [ tport20 ],
+                                        "tagged":   [ "br0", tport22 ]
+                                    },
+                                    {
+                                        "vid": 20,
+                                        "untagged": [ tport21 ],
+                                        "tagged":   [ "br0", tport22 ]
+                                    }
+                                ]
+                            }
+                        }
+                    },
+                    {
+                        "name": tport20,
+                        "type": "iana-if-type:ethernetCsmacd",
+                        "enabled": True,
+                        "infix-interfaces:bridge-port": {
+                            "pvid": 10,
+                            "bridge": "br0"
+                        }
+                    },
+                    {
+                        "name": tport21,
+                        "type": "iana-if-type:ethernetCsmacd",
+                        "enabled": True,
+                        "infix-interfaces:bridge-port": {
+                            "pvid": 20,
+                            "bridge": "br0"
+                        }
+                    },
+                    {
+                        "name": tport22,
+                        "type": "iana-if-type:ethernetCsmacd",
+                        "enabled": True,
+                        "infix-interfaces:bridge-port": {
+                            "bridge": "br0",
+                        }
+                    }
+                ]
+            }
+        })
+
+    with test.step("Ping host:data20 with IP 10.0.0.3 and host:data21 with IP 10.0.0.4 from host:data11 with IP 10.0.0.2"):
+        
+        _, hport10 = env.ltop.xlate("host", "data10")
+        _, hport11 = env.ltop.xlate("host", "data11")
+        _, hport20 = env.ltop.xlate("host", "data20")
+        _, hport21 = env.ltop.xlate("host", "data21")
+
+        with infamy.IsolatedMacVlan(hport10) as ns10, \
+             infamy.IsolatedMacVlan(hport11) as ns11, \
+             infamy.IsolatedMacVlan(hport20) as ns20, \
+             infamy.IsolatedMacVlan(hport21) as ns21:
+
+            ns10.addip("10.0.0.1")
+            ns11.addip("10.0.0.2")
+            ns20.addip("10.0.0.3")
+            ns21.addip("10.0.0.4")
+
+            ns10.ping("10.0.0.3")
+            ns11.ping("10.0.0.4")
+
+            if not ns10.ping_neg("10.0.0.4"):
+                test.fail()
```

Comment:
I'm not smart enough to parse double negations, sorry :sweat_smile: 

I think we can come up with something more readable, see my comment on `netns.py`

Reply:
much more readable!